lms/djangoapps/lms_xblock/mixin.py

```diff
@@ -150,11 +208,17 @@ def validate(self):
         validation = super(LmsBlockMixin, self).validate()
         has_invalid_user_partitions = False
         has_invalid_groups = False
+
         for user_partition_id, group_ids in self.group_access.iteritems():
             try:
                 user_partition = self._get_user_partition(user_partition_id)
             except NoSuchUserPartitionError:
                 has_invalid_user_partitions = True
+            except AttributeError:
+                try:
+                    user_partition = self.user_partitions[user_partition_id]
```

Comment:
I don't understand line 219, as line 214 already attempted to call self.user_partitions[user_partition_id].

When is a KeyError thrown?